python_modules/dagster/dagster/_core/definitions/decorators/decorator_assets_definition_builder.py

```diff
@@ -79,6 +79,7 @@ def build_named_ins(
     fn: Callable[..., Any],
     asset_ins: Mapping[str, AssetIn],
     deps: Optional[AbstractSet[AssetKey]],
+    allow_ins_and_deps_coexistence: bool = False,
```

Comment:
no strong objection here. but i'm away from the codebase for too long so im wondering how @jamiedemaria feels about this approach, in case of any caveat/implication of loosening this.